### DISCUSSION TITLE
Trigger event before any http request

### DISCUSSION THREAD

**Discussion by frenci8** [symfony/symfony#42898] [Q&A] [answered]
Hi,

I am using Symfony 5 and I would like to get the current IP of the user before each http request and detect from that the location of the user and save it in a session so I can use it in the controllers and in twig in the same request without refreshing the page. I know how to get the IP and location of the user, but I can not find how to call those functions in symfony before each http request. 

I can do the same thing by repeating that code in every controller where I need to get the user's location, but that way, there's a lot of repeated code. 

Any ideas how I can create the event? 

Thank you

**Comment by michaoj** [comment, marked as answer]
I think you can use the existing Kernel request event

see : https://symfony.com/doc/current/reference/events.html#kernel-request

**Reply by chyshyya** [reply]
`php bin/console make:subscriber`
Choose `kernel.request`
And do what you need on each request
Make @michaoj answer as ANSWERED. He is absolutely right
